credentials/java/com/google/auth/Credentials.java

```diff
@@ -21,13 +22,25 @@
   /**
    * Get the current request metadata, refreshing tokens if required.
    *
+   * <p>For temporary compatibility only. Passes a null URI.
+   *
+   * @throws IOException if there was an error getting up-to-date access.
+   */
+  public Map<String, List<String>> getRequestMetadata() throws IOException {
+    return getRequestMetadata(null);
+  }
+
+  /**
+   * Get the current request metadata, refreshing tokens if required.
+   *
    * <p>This should be called by the transport layer on each request, and the data should be
    * populated in headers or other context. The operation can block and fail to complete and may do
    * things such as refreshing access tokens.
    *
+   * @param uri URI of the entry point for the request.
    * @throws IOException if there was an error getting up-to-date access.
    */
-  public abstract Map<String, List<String>> getRequestMetadata() throws IOException;
+  public abstract Map<String, List<String>> getRequestMetadata(URI uri) throws IOException;
```

Comment:
There was a lot of discussion about this w.r.t. the C++ implementation, where that pattern was proposed and it created pretty significant difficulties for that implementation. Another motivation is that we can improve the Http experience with this, by looking up and caching scopes from discovery if they are missing.

However, if this model creates difficulty for Java GRPC, it would be possible to support both the parameter model and constructor model. Credentials like JwtAccess that need this can also provide a way to specify a default value for the URI in the constructor and a cloning method.

If this sounds good, I can just revise the documentation to clarify that the parameter free overload is not temporary, and add this behavior into the JwtAccess credentials implementation.